shopinvader/components/service_context_provider.py

```diff
@@ -43,9 +43,14 @@ def _get_backend(self):
         website_unique_key = self.request.httprequest.environ.get(
             "HTTP_WEBSITE_UNIQUE_KEY"
         )
-        return self.env["shopinvader.backend"]._get_from_website_unique_key(
-            website_unique_key
-        )
+        if website_unique_key:
+            return self.env["shopinvader.backend"]._get_from_website_unique_key(
+                website_unique_key
+            )
+        backend = self.env["shopinvader.backend"].search([], limit=2)
+        if len(backend) == 1:
```

Comment:
Thinking twice, I have a question: this way you return the backend even if you don't have anything to validate if you can use the backend. No?

Reply:
https://github.com/shopinvader/odoo-shopinvader/blob/0d835f50127cfe1656758a84ff80321caa771108/shopinvader/models/shopinvader_backend.py#L619

_get_from_website_unique key does not do any validating (except for record rules)

In this case this is the same.

It could cause problem if _get_from_website_unique_key is inherited and a validation is added in it